src/Symfony/Component/DependencyInjection/Loader/PhpFileLoader.php

```diff
@@ -40,13 +40,21 @@ class PhpFileLoader extends FileLoader
 {
     protected bool $autoRegisterAliasesForSinglyImplementedInterfaces = false;
 
+    private ?ConfigBuilderGeneratorInterface $generator;
+
     public function __construct(
         ContainerBuilder $container,
         FileLocatorInterface $locator,
         ?string $env = null,
-        private ?ConfigBuilderGeneratorInterface $generator = null,
+        ConfigBuilderGeneratorInterface|bool|null $generator = null,
         bool $prepend = false,
     ) {
+        if (\is_bool($generator)) {
+            $prepend = $generator;
+            $generator = null;
+        }
```

Comment:
should we trigger a deprecation in the `else` branch ?

Reply:
The issue is that we should still support config builders, so we'd always trigger that one (unless we go with an extra silencing argument, but as I wrote the other comment, this might be good enough in practice.)

Reply:
We can trigger a deprecation in 8.1